### PR TITLE
[jenkins] Moved data source from openshift to jenkins

### PR DESCRIPTION
**Pull request by mocenas**
Changed rourka to get data from jenkins instead from openshift.
Added possibility to run builds from rourka

### CODE REVIEW

File: dashboard/src/main/java/com/github/ladicek/rourka/BuildResource.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    private JenkinsBuildRunner jenkinsBuildRunner;
+
+    @Inject
+    @JenkinsAuthorizedHttpClient
+    private CloseableHttpClient httpClient;
+
+    @GET
+    @Produces("text/plain;charset=utf-8")
+    public String get(@PathParam("name") String buildName) throws Exception {
+        jenkinsBuildRunner = new JenkinsBuildRunner(new JenkinsRequestHandler(httpClient));
```
Comment (Ladicek) [inline dashboard/src/main/java/com/github/ladicek/rourka/BuildResource.java]: Shouldn't be a field, just a local variable.

File: dashboard/src/main/java/com/github/ladicek/rourka/jenkins/JenkinsAuthorizedHttpClientProducer.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+                .build();
+
+        CredentialsProvider credentialsProvider = new BasicCredentialsProvider();
+        credentialsProvider.setCredentials(
+                new AuthScope(AuthScope.ANY_REALM, AuthScope.ANY_PORT),
+                new UsernamePasswordCredentials(jenkinsCredentials.getLogin(), jenkinsCredentials.getApiToken()));
+
+
+        String authStr = jenkinsCredentials.getLogin() +":"+  jenkinsCredentials.getApiToken();
+        String encodedCredentials = DatatypeConverter.printBase64Binary(authStr.getBytes("utf-8"));
```
Comment (Ladicek) [inline dashboard/src/main/java/com/github/ladicek/rourka/jenkins/JenkinsAuthorizedHttpClientProducer.java]: Use `java.util.Base64`.

File: dashboard/src/main/java/com/github/ladicek/rourka/jenkins/JenkinsDataProvider.java [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+    {
+        JsonValue jsonValue=object.get(attribute);
+        if(jsonValue.getValueType().equals(JsonValue.ValueType.NULL)){
+            return "";
+        } else {
+            return object.getString(attribute);
+        }
+    }
+
+    private Boolean readBooleanOrNull(JsonObject object, String attribute)
```
Comment (Ladicek) [inline dashboard/src/main/java/com/github/ladicek/rourka/jenkins/JenkinsDataProvider.java]: These two methods have very similar names, but treat `null` values very differently.

File: dashboard/src/main/java/com/github/ladicek/rourka/jenkins/JenkinsProperties.java [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+public class JenkinsProperties {
+
+    private static Properties properties;
+
+    /*
+     Open and load a properties file
+     */
+    static {
+        try {
+            ClassLoader loader = JenkinsProperties.class.getClassLoader();
```
Comment (Ladicek) [inline dashboard/src/main/java/com/github/ladicek/rourka/jenkins/JenkinsProperties.java:16]: This won't work. I will want to store the credentials into an OpenShift secret, which will be exposed either as environment variables or a file. Probably a file.

File: dashboard/src/main/java/com/github/ladicek/rourka/jenkins/JenkinsRequestHandler.java [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+import java.io.StringReader;
+
+public class JenkinsRequestHandler {
+    private Executor executor;
+
+    @Inject
+    @JenkinsAuthorizedHttpClient
+    private CloseableHttpClient httpClient;
+
+    public JenkinsRequestHandler(HttpClient httpClient) throws Exception{
```
Comment (Ladicek) [inline dashboard/src/main/java/com/github/ladicek/rourka/jenkins/JenkinsRequestHandler.java]: Why do you `@Inject` the HTTP client above, then?

File: dashboard/src/main/resources/jenkins.properties [hunk 6]
```diff
@@ -0,0 +1,5 @@
+jenkins.url=http://jenkins
+
+jenkins.login=xpaasqe-admin
+jenkins.apiToken=3bf6ab02b4928a54c043cbb14483199c
```
Comment (Ladicek) [inline dashboard/src/main/resources/jenkins.properties]: Don't publish secrets! Delete this file from the PR and if this is real token, contact someone to invalidate it right away.

**Review by Ladicek — CHANGES_REQUESTED** [review]
Please fix the code style, too. (Opening braces always on the same line, spaces around assignment operator, etc.)

Also please cut the crap (`AbstractValueTransfer`???), especially when it comes to mutable objects. No JavaBeans.

Otherwise mostly look good. I will merge this when I'm finished with the refactoring -- the dashboard comes last, even though you were able to finish it first :-)

**Comment by Ladicek** [comment]
Oh, one more thing. I don't recall if the code already deals with it or not, but there will be Jenkins jobs _without_ the JSON in description. Some kind of "support" jobs. That's okay and expected; these jobs should simply be skipped, they won't be reported in the dashboard.

**Comment by Ladicek** [comment]
Actually, one other question: the original approach of authenticating to Jenkins using the service account's token worked just fine. Why did you change it?

**Comment by mocenas** [comment]
I changed the authentication because the authenticating thought service account's token logged user in as a service account -> which has read only access so it cannot run builds. 
I was unable to change priviledge to a service account so I changed auth. to be able to run builds

**Comment by Ladicek** [comment]
Hmm, makes sense. I think it should be doable, but let's keep your change for now. I'll investigate later.

**Comment by mocenas** [comment]
Well am sorry. I already investigated it deeper and found out that it is doable and I done it. So in my new version it is again done by token authentication.

**Comment by Ladicek** [comment]
Nice, thanks! :-)

**Comment by mocenas** [comment]
Ok I changed indentation from tabs to spaces

**Comment by Ladicek** [comment]
I've merged this locally and did a significant refactoring. Closing this PR, but none of this is actually dropped. Thanks!

**Pull request closed without merging**